ui/changes/src/feature.js

```diff
@@ -1,5 +1,73 @@
 import * as common from "./common.js";
 
+var all_filters_ids = [
+  "releaseVersions_feature_html",
+  "types_feature_html",
+  "testingTags_feature_html",
+  "severities_feature_html",
+  "riskiness_feature_html",
+  "products_feature_html",
+  "components_feature_html",
+];
+var k = {};
+for (var i = 0; i < all_filters_ids.length; i++) {
+  k[all_filters_ids[i]] = 0;
+}
+var button_prefix = ["", "Undo "];
+
+function selectAll(el_id) {
+  // get select element
+  var el;
+
+  var s_index = el_id.search("_");
+  if (s_index != -1) {
+    el = document.getElementById(String(el_id.slice(0, s_index)));
+  } else {
+    el = document.getElementById(el_id);
+  }
+
+  for (var p = 0; p < el.options.length; p++) {
+    if (!k[el_id]) {
+      el.options[p].selected = true;
+    } else {
+      el.options[p].selected = false;
+    }
+  }
+  k[el_id] = !k[el_id];
+
+  var select_all_el = document.getElementById("select_all_" + el_id);
+  select_all_el.textContent = button_prefix[Number(k[el_id])] + "Select All";
+}
+
+document.getElementById("select_all_" + all_filters_ids[0]).onclick =
+  function () {
+    selectAll(all_filters_ids[0]);
+  };
+document.getElementById("select_all_" + all_filters_ids[1]).onclick =
+  function () {
+    selectAll(all_filters_ids[1]);
+  };
+document.getElementById("select_all_" + all_filters_ids[2]).onclick =
+  function () {
+    selectAll(all_filters_ids[2]);
+  };
+document.getElementById("select_all_" + all_filters_ids[3]).onclick =
+  function () {
+    selectAll(all_filters_ids[3]);
+  };
+document.getElementById("select_all_" + all_filters_ids[4]).onclick =
+  function () {
+    selectAll(all_filters_ids[4]);
+  };
+document.getElementById("select_all_" + all_filters_ids[5]).onclick =
+  function () {
+    selectAll(all_filters_ids[5]);
+  };
+document.getElementById("select_all_" + all_filters_ids[6]).onclick =
+  function () {
+    selectAll(all_filters_ids[6]);
+  };
```

Comment:
Most of the code is duplicated between the various JS files. We should instead share it in common.js so it's automatically loaded by all.

Reply:
Done.